src/stream/mod.rs

```diff
@@ -67,6 +114,10 @@ pub struct Stream {
     pub video_details: Arc<VideoDetails>,
     #[derivative(Debug = "ignore", PartialEq = "ignore")]
     client: Client,
+    #[cfg(any(feature = "callback", doc))]
+    #[doc(cfg(feature = "callback"))]
+    #[derivative(Debug = "ignore", PartialEq = "ignore")]
+    pub callback: Option<Callback>,
```

Comment:
I thougth about directly supplying those closures to the download functions, but this would mean that we need a new set of ` download`, `download_to_dir`, `download_to`, `blocking_download`, `blocking_download_to_dir`, `blocking_download_to` that accepts on_progress or on_complete types. I'd be happy to find a solution that doesn't involve that much more double code.

Reply:
I also thought about adding slow variants of connect_on_progress\* that only get executed every 100th time an event occurs.
<details>
<summary>I just checked how large those chunks actually are by fetching the wikipedia logo.</summary>
Chunk len: 3600<br/>
Chunk len: 8192<br/>
Chunk len: 8191<br/>
Chunk len: 16383<br/>
Chunk len: 2<br/>
Chunk len: 16383<br/>
Chunk len: 11874<br/>
Chunk len: 4511<br/>
Chunk len: 16383<br/>
Chunk len: 16383<br/>
Chunk len: 2<br/>
Chunk len: 16383<br/>
Chunk len: 16383<br/>
Chunk len: 2<br/>
Chunk len: 16383<br/>
Chunk len: 16383<br/>
Chunk len: 2<br/>
Chunk len: 16383<br/>
Chunk len: 16383<br/>
Chunk len: 2<br/>
Chunk len: 16383<br/>
Chunk len: 16383<br/>
Chunk len: 2<br/>
Chunk len: 16383<br/>
Chunk len: 16383<br/>
Chunk len: 2<br/>
Chunk len: 16383<br/>
Chunk len: 16383<br/>
Chunk len: 2<br/>
Chunk len: 16383<br/>
Chunk len: 16383<br/>
Chunk len: 2<br/>
Chunk len: 16383<br/>
Chunk len: 16383<br/>
Chunk len: 2<br/>
Chunk len: 4313<br/>
</details>
So fetching once every 100 events will trigger the function roughly every MB downloaded.